zipline/pipeline/factors/factor.py

```diff
@@ -1205,17 +1205,22 @@ def compute(self, today, assets, out, close):
     '''
     dtype = float64_dtype
 
-    def __getattr__(self, attribute_name):
+    def __getattribute__(self, name):
+        outputs = object.__getattribute__(self, 'outputs')
+        if outputs is not NotSpecified and name in outputs:
+            return RecarrayField(factor=self, attribute=name)
+        else:
+            return super(CustomFactor, self).__getattribute__(name)
+
+    def __getattr__(self, name):
```

Comment:
I think it's probably slightly cleaner just because it keeps all the logic in one place.  I could go either way on this though, so do what you think makes sense.